File: java-services/src/main/java/com/appdynamics/lambda_demo/actions/Action.java
package com.appdynamics.lambda_demo.actions;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Map;

public abstract class Action {
    
    String name = "";
    transient Map<String, Object> properties = null;
    String type = "";

    public transient final Logger logger = LoggerFactory.getLogger(Action.class);

    public Action(String name, Map<String, Object> properties) {
        this.name = name;
        this.properties = properties;
    }

    public abstract void execute();

}

File: nodejs-services/src/actions/action.js
const Logger = require('../logger');

class Action { 
    constructor(name, props) {
        if (!this.execute) {
            throw new Error("Execute function not implemented!");
        }

        this.name = name;
        this.props = props;
        this.type = "";
        this.logger = new Logger().getInstance();
    }
}

module.exports = Action;